lib/AST/ASTPrinter.cpp

```diff
@@ -5112,8 +5137,7 @@ void PrintAST::visitTupleExpr(TupleExpr *expr) {
   Printer << ")";
 }
 
-void PrintAST::visitTypeJoinExpr(TypeJoinExpr *expr) {
-}
+void PrintAST::visitTypeJoinExpr(TypeJoinExpr *expr) { }
```

Comment:
Might be worth adding an `llvm_unreachable` in the body of this, since we should never see a TypeJoinExpr outside of the constraint system.

Reply:
Cool.